src/SolverAppGateway.sol

```diff
@@ -41,16 +78,19 @@ contract SolverAppGateway is AppGatewayBase {
      * @dev Sets up the validity of the deployed OnchainTrigger contract on the specified chain
      * @param chainSlug_ The identifier of the chain where the contract was deployed
      */
-    function initialize(uint32 chainSlug_) public override {
-        address spokePoolWrapperOnChainAddress = getOnChainAddress(spokePoolWrapper, chainSlug_);
-        address wethVaultOnChainAddress = getOnChainAddress(wethVault, chainSlug_);
-        if (chainSlug_ == BASE_SEPOLIA_CHAIN_ID) {
-            SpokePoolWrapper(spokePoolWrapperOnChainAddress).setSpokePool(address(spokePoolBase));
-            WETHVault(wethVaultOnChainAddress).setSpokePool(address(spokePoolBase));
-        } else {
-            SpokePoolWrapper(spokePoolWrapperOnChainAddress).setSpokePool(address(spokePoolArbitrum));
-            WETHVault(wethVaultOnChainAddress).setSpokePool(address(spokePoolArbitrum));
-        }
+    function initialize(uint32 chainSlug_) public override async {
```

Comment:
just wondering why we even need initialize if we can have the conditional constructor args. we could just set the spoke pool address in the constructor wdyt?